File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        
        
        
        vector<int>res;
        int size=nums.size();
	for(int i=0;i<size;i++)
	{
        
        int k=target-nums[i];
		for(int j=i+1;j<size;j++)
		{
			if(nums[j]==k)
			{
				res.push_back(i);
				res.push_back(j);
				break;
			}
		}
        if(res.size()==2)
            break;
	}
        
        
        
        
        
        
        return res;
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mp = new Map();
    for (let i in nums) {
        mp.set(nums[i], i);
    }
    
    for (let i in nums) {
    let num = nums[i];
    let numToSearchFor = target - num;
    if (mp.get(numToSearchFor)) {

        if (mp.get(numToSearchFor) != i) {
            return [i, mp.get(numToSearchFor)];
        }
    }
}
    
};